scipy/sparse/csgraph/_laplacian.py

```diff
@@ -34,8 +34,9 @@ def laplacian(csgraph, normed=False, return_diag=False, use_out_degree=False):
 
     Returns
     -------
-    lap : ndarray
-        The N x N laplacian matrix of graph.
+    lap : ndarray or coo_matrix
+        The N x N laplacian matrix of csgraph. It will be a numpy array (dense)
+        if the input was dense, sparse (COO format) otherwise.
```

Comment:
Well, the docstring can be changed in such an (unlikely) eventuality... This code hasn't been touched in a long time. And documenting the correct return type enables users to make their own optimizations.

Having said that, I'm new here, so if you insist I'll change it to read just sparse matrix. =) But that's my 2c.